### PR TITLE
docs: Document SSO expiry option

### LINKED ISSUE #5545: Allow us to customize the sso logout timer
# Summary

If you have SSO set up, you get logged out after 24 hours (possibly before, but I do sleep somtimes!). This is frustrating for our end users who are used to tools of this nature remembering their credentials for a long time after logging in.

However, I appreciate that some companies will have security requirements that mandate that users are logged out more frequently.

An option in the workflow controller configmap (under sso:) to choose the timeout time would be ideal. An option to never expire would also be useful.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by tico24**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Fixes #5545 by documenting the sso expiry option.

### CODE REVIEW

File: USERS.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ Currently, the following organizations are **officially** using Argo Workflows:
 1. [SAS](https://www.sas.com/)
 1. [Schlumberger](https://slb.com/)
 1. [SegmentStream](https://segmentstream.com)
+1. [Sendible](https://sendible.com)
```
Comment (simster7) [inline USERS.md:108]: Welcome!

File: docs/argo-server-sso.md [hunk 2]
```diff
@@ -149,6 +149,7 @@ What this might look like in your chart configuration:
          - --auth-mode=sso
        sso:
          issuer: https://argo-cd.mydomain.com/api/dex
+         expiry: 240
```
Comment (simster7) [inline docs/argo-server-sso.md:152]: 1. Could we add a similar note about the ability to determine the length of your login somewhere in this doc?

1. Could we add a quick inline comment here as to what this does, and its units (hours)

**Review by simster7 — CHANGES_REQUESTED** [review]
Thanks! Two minor comments

File: docs/workflow-controller-configmap.yaml [hunk 3]
```diff
@@ -267,6 +267,9 @@ data:
   sso: |
     # This is the root URL of the OIDC provider (required).
     issuer: https://issuer.root.url/
+    # This defines how long your login is valid for (in hours). (optional)
+    # If omitted, defaults to 24h. Example below is 10 days.
+    expiry: 240
```
[the remark this answers is not in the document]
Comment (tico24) [reply docs/workflow-controller-configmap.yaml:272]: It's a number (from my understanding). Want me to remove the 'h' from '24' to help clarify?

[the remark this answers is not in the document]
Comment (alexec) [reply docs/workflow-controller-configmap.yaml:272]: can you confirm by testing using a number? the source code suggests it is a duration

[the remark this answers is not in the document]
Comment (tico24) [reply docs/workflow-controller-configmap.yaml:272]: I've been using a number all day... obviously I won't know if it's worked until 24h are up :)

[the remark this answers is not in the document]
Comment (tico24) [reply docs/workflow-controller-configmap.yaml]: My coding abilities are poor, but I believe you are right. I have updated accordingly.

**Comment by tico24** [comment]
That said.. setting it to either "1" or "1h" doesn't boot me out after an hour....

**Review by simster7 — APPROVED** [review]

**Comment by simster7** [comment]
> That said.. setting it to either "1" or "1h" doesn't boot me out after an hour....

Bug report?

**Comment by alexec** [comment]
Make sure you restart the server after changing config. If you don’t get booted out, that’s a bug.

**Comment by tico24** [comment]
I have definitely restarted it. Long weekend here in the uk now, but I'll continue to test on Tuesday before raising a bug.

File: docs/argo-server-sso.md [hunk 4]
```diff
@@ -108,6 +108,18 @@ If no rule matches, we deny the user access.
...
+## SSO Login Time
+
+> v2.12 and after
+
+By default, your SSO session will expire after 10 hours. You can change this by adding an expiry value to your [workflow-controller-configmap.yaml](workflow-controller-configmap.yaml) under the SSO heading.
+
+    ```yaml
+    sso:
+      # Expiry defines how long your login is valid for in hours. (optional)
+      expiry: 240h
```
Comment (alexec) [inline docs/argo-server-sso.md:120]: This is `sessionExpiry`.

**Review by alexec — CHANGES_REQUESTED** [review]

File: docs/argo-server-sso.md (hunk 4 above)
Comment (tico24) [reply docs/argo-server-sso.md]: Cheers. I did warn you I'm not great with code!

**Review by alexec — APPROVED** [review]

**Pull request merged**